homeassistant/components/local_calendar/calendar.py

```diff
@@ -107,6 +124,71 @@ async def _async_store(self) -> None:
         content = IcsCalendarStream.calendar_to_ics(self._calendar)
         await self._store.async_store(content)
 
+
+class RemoteCalendarEntity(LocalCalendarEntityBase):
+    """A read-only calendar that we get and refresh from a url."""
+
+    def __init__(
+        self,
+        store: LocalCalendarStore,
+        calendar: Calendar,
+        name: str,
+        unique_id: str,
+        url: str,
+        update_interval: timedelta,
+    ) -> None:
+        """Initialize a remote read-only calendar."""
+        super().__init__(store, calendar, name, unique_id)
+        self._url = url
+        self._client = None
+        self._track_fetch = None
+        self._etag = None
+        self._update_interval = update_interval
+
+    async def _fetch_calendar_and_update(self):
```

Comment:
I imagined this working more like a sync where we fetch the url, writing down to the local filesystem store, but from there the local calendar integration works like normal and is able to serve from the local filesystem store.  This means it continues to work offline and truly is still a local calendar. I think it could fetch and write to the store, then work like normal (but still readonly).

I think this could be a good way to think about business logic separation for the "Store" class and fetching/reading/writing.  (Right now there is a "store" for a remote calendar entity and it is ambiguous what its for so something needs to be addressed here)

Reply:
I understand your proposal and I agree with it. It makes a lot of sense. I will implement the changes and tests and ask for a second review.

Thank you very much for your time